File: 75-sort-colors/sort-colors.go
func sortColors(nums []int)  {
    l, mid, r := 0, 0, len(nums) - 1

    for mid <= r {
        if nums[mid] == 0 {
            nums[mid], nums[l] = nums[l], nums[mid]
            l++
            mid++
        } else if nums[mid] == 2 {
            nums[mid], nums[r] = nums[r], nums[mid]
            r--
        } else {
            mid++
        }
    }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int colors[3];

        for (int i : nums) {
            colors[i]++;
        }

        int begin = 0;
        for (int color = 0; color < 3; color++) {
            for (int i = begin; i < begin + colors[color]; i++) {
                nums[i] = color;
            }
            begin += colors[color];
        }
    }
};